### PR TITLE
[VAOS] Calendar Widget 

### PR DESCRIPTION
**Pull request by narin**
## Description
PR for calendar widget that will be used for direct schedule and appointment request.  This PR is for the calendar widget only.  Connecting it to the new appointment flow will be handled in a different PR (https://github.com/department-of-veterans-affairs/vets-website/pull/10941).

## Testing done
Local and unit tests

### Direct Schedule
Screen Shot 2019-10-10 at 11 00 37 PM

### Request
image

## Acceptance criteria
- [x] Calendar only shows available dates as clickable
- [x] Calendar shows unavailable dates as greyed out
- [x] In direct scheduling, once date selected, available times expand beneath
- [x] In request flow, once date is selected, am / pm option expands beneath (confirm this behavior with @rtwell )
- [x] Calendar shows dates and times in facility timezone (confirm this as well)
- [x] Calendar shows available dates up to 45 days in advance (this specific number may change)
- [x] Calendar shows available dates for two stacked months at a time
- [x] Calendar includes 'next' / 'previous' buttons that allow veterans to advance months
- [x] Calendar looks like [current mocks]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/vaos/containers/DateTimeSelectPage.jsx [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+} from '../actions/newAppointment.js';
+import { getFormPageInfo } from '../utils/selectors';
+import slots from './../api/slots.json';
+import Calendar from './../components/calendar/CalendarWidget';
+import { focusElement } from 'platform/utilities/ui';
+import moment from 'moment-timezone';
+
+const pageKey = 'dateTimeSelect';
+
+const institutionTimezone = 'America/Denver';
```
Comment (narin) [inline src/applications/vaos/containers/DateTimeSelectPage.jsx]: Temporarily hardcoded this timezone.  Will be grabbing this from selected institution/clinic.

File: src/applications/vaos/containers/DateTimeSelectPage.jsx [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+          maxSelections={1}
+          getSelectedDateOptions={this.getSelectedDateOptions}
+        />
+      </div>
+    );
+  }
+}
+
+function mapStateToProps(state) {
+  return { ...getFormPageInfo(state, pageKey), availableSlots: slots };
```
Comment (narin) [inline src/applications/vaos/containers/DateTimeSelectPage.jsx]: `availableSlots` will come from redux state instead of being hardcoded like this.

**Comment by jbalboni** [comment]
@narin Looks good so far. This is what I've noticed from playing around with it:

1. I'm not sure if this is properly called out in the ticket, but for requested dates, the AM/PM choice should be checkboxes, so that you can select both at once. This could be something we change in a later ticket, but Ryan and I had a discussion about it and since you can choose the same two dates with AM and PM on the current app, we need to allow that here.
2. The hidden buttons need `aria-hidden` on them, since they're still in the markup.
3. The buttons probably need `aria-label` attributes that have a more descriptive label than just the day of the month that's in the button text.
4. I'd wait for @1Copenut to weigh in on this before changing anything, but I suspect we want the tab order to from the day you've currently selected, to the time radio buttons, then to the next day. Currently, you have to tab through all the days in single row to get to the time radio buttons.
5. We might want `aria-expanded` to be set to true on buttons that are selected and open. This is sort of related to item 4, so Trevor would know for sure.
6. Not sure if you've looked into it, but we might want to figure out how to make sure we're not pulling in more than what we need from moment-timezone. It looks like the size of it varies depending on what you need.
7. I wonder if we want to disable the Next button once we've hit the end of the available appointment slots or at some kind of limit for requested dates? I think there's a max we can schedule out.

**Comment by narin** [comment]
@jbalboni All great points. Thanks for taking a look.

**Comment by narin** [comment]
Hey @jbalboni I've addressed all your suggestions aside from the `moment-timezone` one.  The NPM package seems to be an all or nothing type situation. Working on finding a more lightweight replacement for it.

In regards to the AM/PM checkboxes, initially I was using an object to keep track of selected dates, so that I could easily look up `selectedDates[date]`.  This worked fine when there was only 1 possible additional option (radio buttons). However with the checkbox requirement for AM/PM, I modified the code to use an array to store the selected combinations.  This required some refactoring and the addition of a couple helper functions.

Please take a look and review if the UI portion of this looks good so we can get it merged and @1Copenut can start taking a look.  Will create a separate PR to hook up to the SchemaForm/new appointment flow. Thanks.

**Review by jbalboni — APPROVED** [review]

**Review by 1Copenut — COMMENTED** [review]
I posted some 508 items and a Codepen mockup in the following issue comment: https://github.com/department-of-veterans-affairs/va.gov-team/issues/2569#issuecomment-544709696

**Comment by 1Copenut** [comment]
@narin I talked with @markgreenburg this afternoon. I'd say if you can get the `roles` added as part of this PR, let's do that, and I'll write specific issue tickets for the axe and keyboard items under the https://app.zenhub.com/workspaces/vsp-5cedc9cce6e3335dc5a49fc4/issues/department-of-veterans-affairs/va.gov-team/2671 epic.

**Review by rianfowler — APPROVED** [review]

**Pull request merged**